### PR TITLE
Added guide for token gated rooms on Solana

### PR DESCRIPTION
**Pull request by vrajdesai78**

### CODE REVIEW

File: pages/guides/solana-token-gated.mdx [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+import React, { useMemo } from 'react';
+
+const WalletContextProvider: FC<{ children: ReactNode }> = ({ children }) => {
+    const network = WalletAdapterNetwork.Mainnet;
+
+    const endpoint = useMemo(() => clusterApiUrl(network), [network]);
+
+    const wallets = useMemo(
+        () => [new PhantomWalletAdapter(), new SolflareWalletAdapter({ network }), new TorusWalletAdapter()],
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (mizanxali) [inline pages/guides/solana-token-gated.mdx:94]: rm this eslint-disable

File: pages/guides/solana-token-gated.mdx [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+            size: 8,
+        },
+    },
+});
+
+const App: FC<AppProps> = ({ Component, pageProps }) => {
+    return (
+        <>
+            <Head>
+                <title>Solan Token Gated Example</title>
```
Comment (mizanxali) [inline pages/guides/solana-token-gated.mdx:146]: solana

File: pages/guides/solana-token-gated.mdx [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+```tsx filename="pages/[roomId].tsx"
+import { useWallet } from '@solana/wallet-adapter-react';
+import { useEffect } from 'react';
+import { useRoom } from '@huddle01/react/hooks';
+
+const ReactUIWalletMultiButtonDynamic = dynamic(
+    async () => (await import('@solana/wallet-adapter-react-ui')).WalletMultiButton,
+    { ssr: false }
+);
+
```
Comment (mizanxali) [inline pages/guides/solana-token-gated.mdx:387]: just add the main component declaration line

```
export default function Home() {
// rest of the code
}
```

**Review by mizanxali — CHANGES_REQUESTED** [review]
added comments

**Comment by vrajdesai78** [comment]
> added comments

Pushed required changes

**Review by mizanxali — APPROVED** [review]

**Pull request merged**